accepted/0000-workspaces-part2.md

```diff
@@ -0,0 +1,196 @@
+# npm workspaces: Running commands
+
+## Summary
+
+Introduces basic support to running **npm commands** across nested **workspaces**.
+
+## Motivation
+
+The need for running **npm commands** across nested **workspaces** was sourced from the community during the ellaboration of the [original **npm workspaces** RFC](https://github.com/npm/rfcs/pull/103).
+
+## Detailed Explanation
+
+Following up from the original **npm workspaces** RFC are the following command-related changes:
+
+- A subset of the standard **npm commands** needs to be made aware of **npm workspaces**
+- It's desired to have a standard way to filter out **workspaces** in which to run specific **npm commands**
+
+## Rationale and Alternatives
+
+The ability to run **npm commands** across defined **workspaces** is essential to successfully manage a **npm workspaces** workflow. That makes the following alternatives much less desireable:
+
+- Do not implement further support to manage running commands across **workspaces**
+- Implement only _some_ of the implementation changes described in this document
+
+**Note:** Alternative ways of executing each of the described features are defined along with their implementation details below.
+
+## Implementation
+
+### 1. Run commands across all child packages
+
+Make **npm cli** subcommands **npm workspaces**-aware, so that running a command tries to run it within all **workspaces** as long as a **workspaces configuration** field is properly defined in `package.json`.
+
+Only a subset of commands are to be supported:
```

Comment:
yeah, I think what we need here is a good definition of what is a package-related command vs all the other administrative kind of commands

it makes sense to suppport `--workspaces=foo` in commands such as `npm test`, `npm ls` but it makes no sense in admin types of commands such as `npm adduser`, `npm token`, etc

so that was my goal here with the "subset of commands" definition 😊 I'm def open to reword that!

Reply:
revamped all this session and categorized **all** top-level commands to make sure the support matrix is extra clear 😊